src/geometry/ArborX_Ray.hpp

```diff
@@ -112,7 +113,30 @@ bool intersection(Ray const &ray, Box<3> const &box, float &tmin, float &tmax)
   {
     float tdmin;
     float tdmax;
-    if (dir[d] >= 0)
+    if (dir[d] == 0.)
+    {
+      float const min_orig = min[d] - orig[d];
+      float const max_orig = max[d] - orig[d];
+
+      // minx_orig is zero then max_orig is also zero
+      if (min_orig == 0.)
+      {
+        continue;
+      }
+
+      // signbit returns false for +0.0 and true for -0.0
+      if (std::signbit(dir[d]))
+      {
+        tdmin = std::signbit(max_orig) ? inf : -inf;
+        tdmax = std::signbit(min_orig) ? inf : -inf;
+      }
+      else
+      {
+        tdmin = std::signbit(min_orig) ? -inf : inf;
+        tdmax = std::signbit(max_orig) ? -inf : inf;
+      }
```

Comment:
```suggestion
      tdmin = (std::signbit(dir[d]) == std::signbit(max_orig) ? inf : -inf;
      tdmax = (std::signbit(dir[d]) == std::signbit(min_orig) ? inf : -inf;
```
I think

Reply:
Do you care about this change? I find it harder to reason about .

Reply:
I can see your point.
What about 
```suggestion
      tdmin = std::signbit(dir[d] * max_orig) ? inf : -inf;
      tdmax = std::signbit(dir[d] * min_orig) ? inf : -inf;
```
Essentially, whether they are pointing in the same direction?